src/cirrus/package.py

```diff
@@ -0,0 +1,388 @@
+        help="initial package version",
+        default='0.0.0',
+        dest='version'
+    )
+    init_command.add_argument(
+        '--organization', '-o',
+        dest='org',
+        default='ORGANIZATION HERE',
+    )
+    init_command.add_argument(
+        '--description', '-d',
+        dest='desc',
+        default='PACKAGE DESCRIPTION HERE'
+    )
+
+    init_command.add_argument(
+        '--templates',
+        help='template rules to include in MANIFEST',
+        nargs='+',
+        default=list()
+    )
+    init_command.add_argument(
+        '--version-file',
+        help='Version file, defaults to package __init__.py',
+        default=None
+    )
+    init_command.add_argument(
+        '--history-file',
+        help='changelog history filename',
+        default='HISTORY.md'
+    )
+    init_command.add_argument(
+        '--requirements',
+        help='requirements file for pip',
+        default='requirements.txt'
+    )
+    init_command.add_argument(
+        '--master-branch',
+        help='GitFlow master branch',
+        default='master',
+        dest='master'
+    )
+    init_command.add_argument(
+        '--develop-branch',
+        help='GitFlow develop branch',
+        default='develop',
+        dest='develop'
+    )
+    init_command.add_argument(
+        '--no-remote',
+        help='disable pushing changes to remote, commit locally only',
+        default=False,
+        action='store_true'
+    )
+
+    subl_command = subparsers.add_parser('sublime-project')
+    subl_command.add_argument(
+        '--project-name', '-p',
+        dest='project',
+        help='Name of project, defaults to name of cirrus package',
+        default=None
+    )
+    subl_command.add_argument(
+        '--python-path',
+        nargs='+',
+        help='extra directories in package to add to pythonpath in sublime project',
+        dest='pythonpath',
+        default=list()
+    )
+
+    opts = parser.parse_args(argslist)
+    return opts
+
+
+@contextlib.contextmanager
+def working_dir(new_dir):
+    """
+    helper to switch to repo dir and back
+    """
+    cwd = os.getcwd()
+    try:
+        os.chdir(new_dir)
+        yield new_dir
+    finally:
+        os.chdir(cwd)
+
+
+def setup_branches(opts):
+    """
+    set up git branches, starting from master
+    """
+    do_push = not opts.no_remote
+    LOGGER.info(
+        "setting up branches master={} develop={}".format(
+            opts.master, opts.develop
+        )
+    )
+    with working_dir(opts.repo):
+        branch(opts.repo, opts.master, opts.master)
+        branch(opts.repo, opts.develop, opts.master)
+        if do_push:
+            LOGGER.info("Pushing {}...".format(opts.develop))
+            push()
+
+    LOGGER.info("Working on {}".format(get_active_branch(opts.repo)))
+
+
+def commit_and_tag(opts, *files):
+    """
+    add files, commit changes and verify that initial tag
+    exists
+
+    """
+    do_push = not opts.no_remote
```

Comment:
While it feels a bit backwards in hindsight its at least consistent with the git tools method in that you explicitly enable push. I may refactor this in the future across all the modules, but for now its consistent at least.